engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSpec.scala

```diff
@@ -73,29 +76,48 @@ class SpelExpressionSpec extends FunSuite with Matchers with EitherValues {
+    parse(expr, validationCtx, Map.empty, flavour, strictMethodsChecking = false, staticMethodInvocationsChecking = true)
   }
 
   private def parse[T: TypeTag](expr: String, context: Context = ctx, flavour: Flavour = Standard): ValidatedNel[ExpressionParseError, TypedExpression] = {
     val validationCtx = ValidationContext(
       context.variables.mapValuesNow(Typed.fromInstance))
-    parse(expr, validationCtx, Map.empty, flavour, strictMethodsChecking = true)
+    parse(expr, validationCtx, Map.empty, flavour, strictMethodsChecking = true, staticMethodInvocationsChecking = true)
   }
 
   private def parse[T: TypeTag](expr: String, validationCtx: ValidationContext): ValidatedNel[ExpressionParseError, TypedExpression] = {
-    parse(expr, validationCtx, Map.empty, Standard, strictMethodsChecking = true)
+    parse(expr, validationCtx, Map.empty, Standard, strictMethodsChecking = true, staticMethodInvocationsChecking = true)
   }
 
   private def parse[T: TypeTag](expr: String, validationCtx: ValidationContext, dictionaries: Map[String, DictDefinition],
-                                flavour: Flavour, strictMethodsChecking: Boolean): ValidatedNel[ExpressionParseError, TypedExpression] = {
+                                flavour: Flavour, strictMethodsChecking: Boolean, staticMethodInvocationsChecking: Boolean): ValidatedNel[ExpressionParseError, TypedExpression] = {
     val imports = List(SampleValue.getClass.getPackage.getName)
     SpelExpressionParser.default(getClass.getClassLoader, new SimpleDictRegistry(dictionaries), enableSpelForceCompile = true,
-      strictTypeChecking = true, imports, flavour, strictMethodsChecking = strictMethodsChecking)(ClassExtractionSettings.Default).parse(expr, validationCtx, Typed.fromDetailedType[T])
+      strictTypeChecking = true, imports, flavour, strictMethodsChecking = strictMethodsChecking, staticMethodInvocationsChecking = true, typeDefinitionSetWithDefaultClasses)(ClassExtractionSettings.Default).parse(expr, validationCtx, Typed.fromDetailedType[T])
+  }
+
+  private def typeDefinitionSetWithDefaultClasses: TypeDefinitionSet = {
+
+    def createTestClazzDefinitionFromClassNames(className: String) =
+      TypeInfos.ClazzDefinition(TypedClass(ClassUtils.primitiveToWrapper(ClassUtils.getClass(className)), Nil), Map.empty)
+
+    val stringClazzDefinition = createTestClazzDefinitionFromClassNames("java.lang.String")
+    val longClazzDefinition = createTestClazzDefinitionFromClassNames("java.lang.Long")
+
+    TypeDefinitionSet(Set(stringClazzDefinition, longClazzDefinition))
+  }
+
+  test("evaluate static method call on validated class") {
+    parseOrFail[String]("T(java.lang.String).copyValueOf({'t', 'e', 's', 't'})").evaluateSync[String](ctx) should equal("test")
+  }
+
+  test("evaluate static method call on unvalidated class") {
+    parseOrFail[Long]("T(java.lang.Long).parseLong('1')").evaluateSync[Long](ctx) should equal(1L)
```

Comment:
I don't understand this scenario. Why it is "unvalidated class" if this type is defined in type definition set?

Reply:
changed it, sohuld be ok now